### PR TITLE
Add Update plugin method

### PR DESCRIPTION
**Pull request by kargakis**
Fixes https://github.com/openshift/openshift-azure/issues/334

### CODE REVIEW

**Comment by kargakis** [comment]
@openshift/sig-azure

File: pkg/upgrade/update.go [hunk 1]
```diff
@@ -295,16 +295,9 @@ func updateInPlace(ctx context.Context, cs *api.OpenShiftManagedCluster, p api.P
...
 
-func drain(
-	ctx context.Context,
-	cs *api.OpenShiftManagedCluster,
-	p api.Plugin,
-	vmc compute.VirtualMachineScaleSetVMsClient,
-	role api.AgentPoolProfileRole,
-	instanceID, nodeName string,
-) error {
+func (u *simpleUpgrader) delete(ctx context.Context, cs *api.OpenShiftManagedCluster, vmc compute.VirtualMachineScaleSetVMsClient, role api.AgentPoolProfileRole, instanceID, nodeName string) error {
```
Comment (jim-minter) [inline pkg/upgrade/update.go]: not sure the s/drain/delete/ makes it clearer

Comment (kargakis) [reply pkg/upgrade/update.go]: delete is closer than drain to what this is actually doing :)

**Pull request merged**